programming/sr/vision/index.md

```diff
@@ -10,85 +10,69 @@ Vision
 </div>
 
-The `sr.robot` library contains support for detecting libkoki markers with the provided webcam.
+The `sr.robot` library contains support for detecting fiducial markers with the provided webcam.
 Markers are attached to various items in the Student Robotics arena.
 Each marker encodes a number in a machine-readable way, which means that robots can identify these objects.
-For information on which markers codes are which, see the [markers page](/docs/programming/sr/vision/markers).
+For information on which markers codes are which, see the [markers page](./markers).
 
 Using knowledge of the physical size of the different markers and the characteristics of the webcam,
- libkoki can calculate the position of markers in 3D space relative to the camera.
+your robot can calculate the position of markers in 3D space relative to the camera.
 Therefore, if the robot can see a marker that is at a fixed location in the arena,
  a robot can calculate its exact position in the arena.
 
-The `sr.robot` library provides all of this power through a single function, `R.see`:
+Under the hood, the vision system is based on [Zoloto](https://zoloto.readthedocs.io/), a wrapper around [OpenCV's ArUco library](https://docs.opencv.org/4.5.4/d5/dae/tutorial_aruco_detection.html), using the `36H11` marker set from [April](https://april.eecs.umich.edu/software/apriltag)
 
-~~~~~ python
-from sr.robot import *
-R = Robot()
-markers = R.see()
-~~~~~
+[Camera](#camera) {#camera}
+===========================
+
+The interface to the vision system is through the camera, accessible through `R.camera`.
 
-When called, this function takes a photo through the webcam and searches for markers within it.
-It returns a list of `Marker` objects, each of which describes one of the markers that were found in the image.
-A detailed description of the attributes of Marker objects is provided [later in this page](#Marker).
+see
+:   Take a photo through the webcam, and return a list of [`Marker`](#Marker) instances, each of which describes one of the markers that were found in the image.
 
 Here's an example that will repeatedly print out the distance to each arena marker that the robot can see:
 
 ~~~~~ python
-from sr.robot import *
+from sr.robot3 import *
 R = Robot()
 
 while True:
-    markers = R.see()
+    markers = R.camera.see()
     print("I can see", len(markers), "markers:")
 
     for m in markers:
-        if m.info.marker_type == MARKER_ARENA:
-            print(" - Marker #{0} is {1} metres away".format(m.info.code, m.dist))
+        print(" - Marker #{0} is {1} metres away".format(m.id, m.distance))
 ~~~~~
 
-[Choosing Resolution](#ChoosingResolution) {#ChoosingResolution}
--------------------
+see_ids
+:   Take a photo through the webcam, and return a list of marker ids (**not** full `Marker` objects). This doesn't do the same orientation calculations as `see`, and so is much faster to run.
```

Comment:
```suggestion
:   Take a photo through the webcam, and return a list of marker ids (**not** full `Marker` objects). This doesn't do the same orientation and distance calculations as `see`, and so is much faster to run.
```

Reply:
Resolved in fef872be0567164bb78598bcc8c41126b511cb2e, in a slightly different way